src/store/actions/authAction.ts

```diff
@@ -1,7 +1,73 @@
 import actionTypes from '../actionTypes';
+import {
+  OnSubmitLoginType,
+  OnSubmitRegistrationDataType,
+} from '../../types/types';
+import {firebase} from '@react-native-firebase/auth';
+import {Dispatch} from 'redux';
 
 export const setIsLoggedIn = (value: boolean) =>
   ({
     type: actionTypes.auth.LOGIN,
     payload: value,
   } as const);
+
+export const setUser = (user: any) =>
+  ({
+    type: actionTypes.auth.SET_USER,
+    payload: user,
+  } as const);
+
+export const errorMessage = (err: string) =>
+  ({
+    type: actionTypes.auth.SET_ERROR,
+    payload: err,
+  } as const);
+
+export const loadingStatus = (value: boolean) =>
+  ({
+    type: actionTypes.auth.SET_LOADING,
+    payload: value,
+  } as const);
+
+export const OnSubmitRegistration: any =
+  (data: OnSubmitRegistrationDataType) => async (dispatch: Dispatch) => {
+    try {
+      if (data.password === data.confirmPassword) {
```

Comment:
this if statement is redundant. Your Yup form validates this before triggering onSubmit function, so this condition is always true.